firebase-admin/runtime/src/main/java/io/quarkiverse/googlecloudservices/firebase/admin/runtime/authentication/http/DefaultFirebaseIdentityProvider.java

```diff
@@ -67,11 +80,13 @@ public Uni<SecurityIdentity> authenticate(FirebaseAuthenticationRequest request,
      * Authenticates the provided FirebaseToken and creates a SecurityIdentity.
      *
      * @param token The FirebaseToken to be authenticated.
+     * @param rawToken The raw JWT token
      * @return A SecurityIdentity representing the authenticated user or null if authentication fails.
      */
-    public SecurityIdentity authenticate(FirebaseToken token) {
+    public SecurityIdentity authenticate(FirebaseToken token, String rawToken) {
         var builder = QuarkusSecurityIdentity.builder()
-                .setPrincipal(getPrincipal(token));
+                .setPrincipal(getPrincipal(token, rawToken))
+                .addCredential(new TokenCredential(rawToken, "Firebase"));
```

Comment:
As it's used somewhere else, better to use a constant to avoid duplicating in two different places the token credential type.

Reply:
Fixed